### ISSUE TITLE
[BUG] Local cluster v2prov non-functional when feature flag multi-cluster-management=false

### ISSUE THREAD

**Issue by FrankYang0529** [rancher/rancher#38706]
**Rancher Server Setup**
- Rancher version: v2.6.7
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): v1.22.12+rke2r1

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**
When starting rke2 with rancher, helm-operation-xxx pods can't be completed. The `proxy` container is not terminated.

![Screen Shot 2022-08-24 at 1 59 48 PM](https://user-images.githubusercontent.com/4927361/186341989-630b6fb6-f50b-4f4d-b59c-8e5c20f9a6a7.png)
![Screen Shot 2022-08-24 at 1 59 53 PM 1](https://user-images.githubusercontent.com/4927361/186342028-eb7aba41-7f94-492c-923b-6e6a80e5d002.png)

In the rancher pod, it keeps showing:
```
2022/08/24 03:54:04 [ERROR] error syncing 'local': handler rke-control-plane-trigger: Index with name by-cluster does not exist, requeuing
```
rke-control-plan has an error message in condition.
```
configuring bootstrap node(s) custom-51ee9b389b9c: waiting for cluster agent to connect
```
rancher-system-agent also has some error messages.
```
Aug 24 05:43:55 node1 rancher-system-agent[12871]: time="2022-08-24T05:43:55Z" level=error msg="error while appending ca cert to pool for probe kube-controller-manager"
Aug 24 05:43:55 node1 rancher-system-agent[12871]: time="2022-08-24T05:43:55Z" level=error msg="error loading CA cert for probe (kube-scheduler) /var/lib/rancher/rke2/server/tls/kube-scheduler/kube-scheduler.crt: open /var/lib/rancher/rke2/server/tls/kube-scheduler/kube-scheduler.crt: no such file or directory"
Aug 24 05:43:55 node1 rancher-system-agent[12871]: time="2022-08-24T05:43:55Z" level=error msg="error while appending ca cert to pool for probe kube-scheduler"
Aug 24 05:44:00 node1 rancher-system-agent[12871]: time="2022-08-24T05:44:00Z" level=error msg="error loading CA cert for probe (kube-scheduler) /var/lib/rancher/rke2/server/tls/kube-scheduler/kube-scheduler.crt: open /var/lib/rancher/rke2/server/tls/kube-scheduler/kube-scheduler.crt: no such file or directory"
```

**To Reproduce**
1. Update `settings.yaml` in https://github.com/bk201/vagrant-rancherd.
```
kubernetes_version: v1.22.12+rke2r1
rancher_version: v2.6.7
```
2. Run `vagrant up node1`.
3. Run `./kube-env.sh`.

**Result**
Check pods `KUBECONFIG=./kubeconfig kubectl get pods -n cattle-system`. Some pods can't be completed.

**Expected Result**
All helm-operation-xxx pods can be completed.

**Comment by snasovich** [comment]
@FrankYang0529 , is this a regression compared to 2.6.6 ?

**Comment by FrankYang0529** [comment]
Our last rancher version is `v2.6.4-harvester3`.

https://github.com/harvester/harvester-installer/pull/325/files

**Comment by michieldeg** [comment]
We're seeing this same issue in 2.6.8. We did not have this issue in 2.6.6 (we've skipped 2.6.7).
Ive tried to manually change the image that the proxy container runs. It terminates just fine on shell:v0.1.16 (used by Rancher 2.6.6) and v0.1.17, but doesn't terminate when running shell:v0.1.18 (used by Rancher 2.6.8) or v0.1.19.

**Comment by kurld** [comment]
Exact same issue here after 2.6.7 -> 2.6.8 update.

**Comment by Oats87** [comment]
Potentially resolved via: https://github.com/rancher/rancher/pull/38911

Other issue here: https://github.com/rancher/rancher/issues/38834

**Comment by Oats87** [comment]
Closing as a duplicate. If this still occurs after the other issue fixes are merged, please reopen.

**Comment by FrankYang0529** [comment]
Hi @Oats87, I have to open this issue again because I can still reproduce following error on `v2.6-head`.

In the rancher pod, it keeps showing:
```
2022/08/24 03:54:04 [ERROR] error syncing 'local': handler rke-control-plane-trigger: Index with name by-cluster does not exist, requeuing
```
rke-control-plan has an error message in condition.
```
configuring bootstrap node(s) custom-51ee9b389b9c: waiting for cluster agent to connect
```

**Comment by FrankYang0529** [comment]
> @FrankYang0529 - Have you tried testing this against the `release/v2.7` branch?

I tried again for `release/v2.7`, but it still got same error.

**Reproduce steps**
1. Update `settings.yaml` in https://github.com/bk201/vagrant-rancherd.
```
kubernetes_version: v1.22.12+rke2r1
rancher_version: v2.7-head
```
2. Add `features: multi-cluster-management=true` to [`rancherValues`](https://github.com/bk201/vagrant-rancherd/blob/3c446ed5119bd654eea6cf4aad6919f81653b0f3/Vagrantfile#L70).
```
rancherValues:
  features: multi-cluster-management=true
  noDefaultAdmin: false
  bootstrapPassword: #{$settings['rancher_admin_passwd']}
```
3. Run `vagrant up node1`.
4. Run `./kube-env.sh` to get kubeconfig.
5. Run `KUBECONFIG=kubeconfig kubectl get rkecontrolplane -n fleet-local local -o yaml | yq .status.conditions` to see conditions in RkeControlPlan.

**Comment by Oats87** [comment]
Investigating this issue, the original `helm-operation-xxx can't be completed` errors were a red herring to the actual issue at hand.

The root of this issue was namely a few things that changed between 2.6.4 and 2.6.8/2.6.9.
1. We started relying on the `Connected` condition being set on the `clusters.management.cattle.io` object, which was added as part of the v2provisioning framework, see: https://github.com/rancher/rancher/pull/35545 -- The primary issue with this controller is that it was being registered through the `multi-cluster-manager` code path, when it should probably have been registered with the dashboard code path. I corrected this in my PR, so that that controller is always registered/run.
2. The `Connected` condition would never be set on a local cluster, because the `clusterconnected`  controller would short circuit if `cluster.Spec.Internal` was true, which is the case for a local cluster. I have changed the logic of the controller to simply short circuit to setting `Connected` to true, because for what it's worth, the agent is connected.
3. There were two v2prov indexers that were being registered in a place where multi-cluster-manager was required, and I have moved them to being registered in the `Register` function of the controller that defines and needs them.

With these changes applied, a local v2prov cluster is able to be successfully provisioned and managed, even with the feature flag for `multi-cluster-manager=false`.
